libraries/chain/transaction_context.cpp

```diff
@@ -247,7 +247,9 @@ namespace eosio { namespace chain {
                                                  uint64_t packed_trx_prunable_size )
    {
       const transaction& trx = packed_trx.get_transaction();
-      EOS_ASSERT( trx.delay_sec.value == 0, transaction_exception, "transaction cannot be delayed" );
+      if ( is_transient() ) {
+         EOS_ASSERT( trx.delay_sec.value == 0, transaction_exception, "transaction cannot be delayed" );
+      }
```

Comment:
Once `DISABLE_DEFERRED_TRXS_STAGE_2` is activated blocks should not validate that contain a `delayed` trx. We should have a test for this as well.

Reply:
I originally said stage 2, but I think we want this for stage 1. My understanding is stage 1 is stop deferred/delayed trxs and stage 2 is cleanup.

Reply:
You are right. After stage 1, no new deferred trxs are allowed; existing ones are starting to be retired as expired and
they should not reach block validation.
After stage 2, the whole system should not have any delayed trxs left.
It's better to assert after stage 1. Will change.